File: js/task2b/S2_FOR_04.js
// S2_FOR_04 - Count occurrences

function countOccurrences(values) {
  const counts = {};
  for (const val of values) {
    if (counts[val] === undefined) {
      counts[val] = 1;
    } else {
      counts[val]++;
    }
  }
  return counts;
}

// tests
console.log(countOccurrences(["a", "b", "a", "c", "a", "b"]));
// { a: 3, b: 2, c: 1 }

console.log(countOccurrences([1, 1, 2, 2, 2, 3]));
// { 1: 2, 2: 3, 3: 1 }

console.log(countOccurrences([]));
// {}

File: python/task2b/S2_FOR_04.py
# S2_FOR_04 - Count occurrences (histogram) - no Counter

def count_occurrences(values):
    counts = {}
    for v in values:
        if v in counts:
            counts[v] += 1
        else:
            counts[v] = 1
    return counts

# Tests
print(count_occurrences(["a", "b", "a", "c", "b", "a"]))
print(count_occurrences([1, 2, 2, 3, 3, 3]))
print(count_occurrences([]))
print(count_occurrences(["x"]))
